pkg/security/ptracer/network_handlers.go

```
@@ -0,0 +1,273 @@
+	}
+
+	socketInfo.BoundToPort = addr.port
+	process.FdToSocket[int32(socketfd)] = socketInfo
+
+	return nil
+}
+
+func handleConnect(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg, regs syscall.PtraceRegs, _ bool) error {
+	addrlen := tracer.ReadArgInt32(regs, 2)
+	addr, err := parseAddrInfo(tracer, process, regs, int32(addrlen))
+
+	if err != nil {
+		return err
+	}
+	socketfd := int32(tracer.ReadArgUint32(regs, 0))
+
+	m, ok := process.FdToSocket[socketfd]
+
+	if !ok {
+		return errors.New("unable to find protocol")
+	}
+
+	msg.Type = ebpfless.SyscallTypeConnect
+	msg.Connect = &ebpfless.ConnectSyscallMsg{
+		MsgSocketInfo: ebpfless.MsgSocketInfo{
+			AddressFamily: addr.af,
+			Addr:          addr.ip,
+			Port:          addr.port,
+		},
+		Protocol: m.Protocol,
+	}
+
+	return nil
+}
+
+func handleSocket(tracer *Tracer, _ *Process, msg *ebpfless.SyscallMsg, regs syscall.PtraceRegs, _ bool) error {
+	socketMsg := &ebpfless.SocketSyscallFakeMsg{
+		AddressFamily: uint16(tracer.ReadArgInt32(regs, 0)),
+	}
+
+	if socketMsg.AddressFamily != unix.AF_INET && socketMsg.AddressFamily != unix.AF_INET6 && socketMsg.AddressFamily != unix.AF_UNIX {
+		return nil
+	}
+
+	protocol := int16(tracer.ReadArgInt32(regs, 1))
+	// This argument can be masked, so just get what we need
+	protocol &= 0b1111
+
+	switch protocol {
+	case unix.SOCK_STREAM:
+		socketMsg.Protocol = unix.IPPROTO_TCP
+	case unix.SOCK_DGRAM:
+		socketMsg.Protocol = unix.IPPROTO_UDP
+	default:
+		return nil
+	}
+
+	msg.Socket = socketMsg
+	return nil
+}
+
+func handleAccept(tracer *Tracer, _ *Process, msg *ebpfless.SyscallMsg, regs syscall.PtraceRegs, _ bool) error {
+	fd := tracer.ReadArgInt32(regs, 0)
+
+	msg.Accept = &ebpfless.AcceptSyscallMsg{
+		SocketFd: fd,
+	}
+
+	return nil
+}
+
+// Handle returns
+func handleAcceptRet(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg, regs syscall.PtraceRegs, _ bool) error {
+	addrlen, _ := tracer.ReadArgInt32Ptr(process.Pid, regs, 2)
```

It would be better to return the error of `ReadArgInt32Ptr` (the tracer could run into a pagefault and fail to read addrlen for example)

---

Addressed